### DISCUSSION TITLE
Avoid proposing updates from `docker/image:2` to `docker/image:20241024`

### DISCUSSION THREAD

**Discussion by felipecrs** [renovatebot/renovate#32128] [Suggest an Idea]
### Tell us more.

Even though `20241024` is an integer and greater than `2`, it generally means that such docker image pushes two different tag styles. One using the semver major, and another using a nightly pattern.

Such nightly pattern can be easily and almost unmistakenly inferred. When detected, Renovate can compare against the style of the current version and avoid proposing an update in case they don't match. For example:

Renovate should propose:

- From `docker/image:2` to `docker/image:3`
- From `docker/image:20241024` to `docker/image:20241025`

But Renovate should NOT propose:

- From `docker/image:2` to `docker/image:20241024`

What do you think?

I am willing to contribute this improvement.

**Comment by rarkins** [comment]
I would prefer to keep it simple like proposed here: https://github.com/renovatebot/renovate/issues/20772

**Reply by felipecrs** [reply]
Agreed!

**Comment by felipecrs** [comment]
Fixed by:

- https://github.com/renovatebot/renovate/pull/38854
